server/pulp/server/db/model/dispatch.py

```diff
@@ -589,3 +592,19 @@ def save_with_set_on_insert(self, fields_to_set_on_insert):
         update = {'$set': stuff_to_update,
                   '$setOnInsert': set_on_insert}
         TaskStatus._get_collection().update({'task_id': task_id}, update, upsert=True)
+
+    @classmethod
```

Comment:
`staticmethod` would work for our purposes, but the signal code in mongoengine expects to send in `cls` which we disregard.